src/components/CustomerQuota/ItemsSelection/AddonsItems.tsx

```diff
@@ -1,13 +1,72 @@
-import React, { FunctionComponent } from "react";
-import { ActivityIndicator } from "react-native";
+import React, { FunctionComponent, useContext, useEffect } from "react";
+import { ActivityIndicator, View } from "react-native";
 import { color } from "../../../common/styles";
+import { AlertModalContext } from "../../../context/alert";
+import { AuthContext } from "../../../context/auth";
+import { CampaignConfigContext } from "../../../context/campaignConfig";
+import { usePastTransaction } from "../../../hooks/usePastTransaction/usePastTransaction";
+import { useTranslate } from "../../../hooks/useTranslate/useTranslate";
+import { TransactionsGroup } from "../TransactionsGroup";
+import {
+  groupTransactionsByCategory,
+  sortTransactionsByCategory,
+} from "../utils";
 
-export const AddonsItems: FunctionComponent = () => {
-  return (
+const BIG_NUMBER = 99999;
+
+export const AddonsItems: FunctionComponent<{
+  ids: string[];
+  isShowAddonItems: boolean;
+  categoryFilter?: string[];
+}> = ({ ids, isShowAddonItems, categoryFilter }) => {
+  const { sessionToken, endpoint } = useContext(AuthContext);
+  const { policies: allProducts } = useContext(CampaignConfigContext);
+  const {
+    pastTransactionsResult: sortedTransactions,
+    loading,
+    error,
+  } = usePastTransaction(ids, sessionToken, endpoint, categoryFilter);
+
+  const { showErrorAlert } = useContext(AlertModalContext);
+  useEffect(() => {
+    if (error) {
+      showErrorAlert(error);
+    }
+  }, [error, showErrorAlert]);
+  const latestTransactionTime: Date | undefined =
+    sortedTransactions && sortedTransactions.length > 0
+      ? sortedTransactions[0].transactionTime
+      : undefined;
+
+  const translationProps = useTranslate();
+  const transactionsByCategoryMap = groupTransactionsByCategory(
+    sortedTransactions,
+    allProducts || [],
+    latestTransactionTime,
+    translationProps
+  );
+
+  const transactionsByCategoryList = sortTransactionsByCategory(
+    transactionsByCategoryMap
+  );
```

Comment:
Noticed that we're performing all these on every render, can you check the number of times this is being called? It may be better to use a `useEffect` such that it only updates when `sortedTransactions` updates